pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md

```diff
@@ -95,3 +95,108 @@ void setMeshTriangles(int meshID, int size, int* vertices);
 void setMeshQuads(int meshID, int size, int* vertices);
 void setMeshTetrahedra(int meshID, int size, int* vertices);
 ```
+
+## Putting it all together
+
+Solvers may give you a range of information regarding vertices and faces.
+Some may give you unique ids for vertices, some only coordinates.
+In this section, we handle some common cases and how to implement them.
+
+### Solver provides IDs
+
+You solver gives each vertex or node a unique identifier.
+This identifier is also available when iterating over faces.
+
+In this case you can save a mapping from Solver ID to preCICE Vertex ID, after defining the vertices.
+When iterating over the faces, get the point identifier, map them to preCICE IDs and use those to define your connectivity.
+
+
+### Solver supports labels or tags
+
+You solver doesn't provide a unique identifier for each vertex or node.
+It does provide a functionality to attach some kind of label or tag to nodes.
+This tag/label is also available when iterating over faces.
+
+In this case you can save preCICE Vertex ID as labels directly in the solver vertices.
+Define the vertices using the preCICE API, then iterate over them and apply the preCICE vertex IDs as labels.
+When iterating over the faces, get the preCICE vertex IDs from the point labels, and use those to define your connectivity.
+
+
+### Solver supports coordinates only
+
+Your solver provides coordinates only or labels are already used for another purpose.
+
+In this case, you need to generate a mapping from coordinates to preCICE vertex ids.
+Due to rounding error, a C++ `std::map` or python `dict` may not be sufficient.
```

Comment:
How are rounding errors related to containers? Linking to some reference in the text (and/or adding more context) would be helpful.